test/files/scalacheck/MutableTreeMap.scala

```diff
@@ -0,0 +1,249 @@
+import java.io._
+
+import org.scalacheck._
+import org.scalacheck.Arbitrary._
+import org.scalacheck.Prop.forAll
+
+import scala.collection.generic.CanBuildFrom
+import scala.collection.mutable
+import scala.util.Try
+
+trait MutableTreeMapGenerators {
+
+  // Note: in scalacheck 1.12.2 tree maps can be automatically generated without the need for custom
+  // machinery
+  def genTreeMap[A: Arbitrary: Ordering, B: Arbitrary]: Gen[mutable.TreeMap[A, B]] = {
+    import org.scalacheck.Gen._
+    for {
+      keys <- listOf(arbitrary[A])
+      values <- listOfN(keys.size, arbitrary[B])
+    } yield mutable.TreeMap(keys zip values: _*)
+  }
+
+  implicit def arbTreeMap[A: Arbitrary: Ordering, B: Arbitrary] = Arbitrary(genTreeMap[A, B])
+}
+
+object MutableTreeMapProperties extends Properties("mutable.TreeMap") with MutableTreeMapGenerators {
```

Comment:
Sounds good to me.

On May 29, 2015 12:27:33 AM GMT+02:00, "Rui Gonçalves" <notifications@github.com> wrote:
>> +trait MutableTreeMapGenerators {
>> +
>> +  // Note: in scalacheck 1.12.2 tree maps can be automatically
>generated without the need for custom
>> +  // machinery
>> +  def genTreeMap[A: Arbitrary: Ordering, B: Arbitrary]:
>Gen[mutable.TreeMap[A, B]] = {
>> +    import org.scalacheck.Gen._
>> +    for {
>> +      keys <- listOf(arbitrary[A])
>> +      values <- listOfN(keys.size, arbitrary[B])
>> +    } yield mutable.TreeMap(keys zip values: _*)
>> +  }
>> +
>> +  implicit def arbTreeMap[A: Arbitrary: Ordering, B: Arbitrary] =
>Arbitrary(genTreeMap[A, B])
>> +}
>> +
>> +object MutableTreeMapProperties extends
>Properties("mutable.TreeMap") with MutableTreeMapGenerators {
>
>Hmm, ok! To check if I understood well, I'll create a separate
>`RedBlackTreeProperties` test class with property checks that adding
>and removing keys in any order will keep the invariants. I'll also add
>`++=` and `--=` property checks for the two existing `TreeMap` test
>classes. That way, I won't even need to make `tree` package-private. Is
>this ok? :)
>
>---
>Reply to this email directly or view it on GitHub:
>https://github.com/scala/scala/pull/4504/files#r31284301

-- 
Sent from my Android device with K-9 Mail. Please excuse my brevity.